### PR TITLE
Iss1156 Unified search

### PR DESCRIPTION
**Pull request by jvwong**
refs #1156 

This is a sketch for a new app service endpoint 'search'. It will coordinate 
  - pathway search: sanitize, send off to pc
  - info box data: tokenize, recognize genes, get entity summaries for interactions app
    - sets the stage for deciding what results based on number of genes (enrichment)

Right now takes in POST with `{"query": "TP53 is the best"`} and returns:

```js
{
  "card": {
    "entities": [
      {
        "namespace": "ncbigene",
        "displayName": "tumor protein p53",
        "localId": "7157",
        "description": "This gene encodes a tumor suppressor ...",
        "aliases": [
          ...
        ],
        "aliasIds": [
          "BCC7",
         ...
        ],
        "xrefLinks": [
          ...
        ]
      }
    ]
  },
  "pathways": [
    {
      "uri": "http://identifiers.org/reactome/R-HSA-3232118",
      "biopaxClass": "Pathway",
      "name": "SUMOylation of transcription factors",
      ...
    },
    ...
  ]
}

```

Then we can remove endpoints for `/api/summary/*`, `/api/pc/*` and just use internally.

### CODE REVIEW

**Comment by jvwong** [comment]
Not a strong one, just trying to have a predictable output when we switch from say few genes (interactions) to many (enrichment). What do you think about this idea which I've seen NCBI use - return object with array of keys that declare the object keys available -
```js
{
  "types": ["pathways", "summaries", ...],
  "pathways": [...],
  "summaries": []
}
```

**Comment by d2fong** [comment]
I guess thats okay, but whats the difference between that and just checking each key manually?

**Comment by d2fong** [comment]
oh but I guess in your approach it means the client doesn't have to know that the 'pathways' and 'summaries' keys exist?

**Comment by jvwong** [comment]
I guess the question is what's the best way to display info box/card content contingently (based on the genes recognized)?

File: src/server/routes/index.js [hunk 1]
```diff
@@ -3,14 +3,14 @@ const router = express.Router();
 const pc = require('./pathway-commons-router');
 const summary = require('./summary');
 
-
+router.use('/api/search', require('./search'));
```
Comment (d2fong) [inline src/server/routes/index.js]: search is currently organized under /api/pc if we want /api/search then we should just scrap /api/pc

Comment (jvwong) [reply src/server/routes/index.js]: Eventually, I was thinking we should remove all other 'search' (`/pc`, `/summary`) endpoints and just use the modules directly within this one. Unless there is a reason to expose those?

**Comment by jvwong** [comment]
That `interactions` key is probably not appropriate for enrichment, as the enrichment map doesn't really describe interactions. Maybe `networks` or ...?

**Comment by d2fong** [comment]
Yeah that makes sense

File: src/server/routes/index.js (hunk 1 above)
Comment (d2fong) [reply src/server/routes/index.js]: Yeah that makes sense

**Comment by jvwong** [comment]
That's not what I expected rebase to do.

**Comment by d2fong** [comment]
did you want to revert it or no?

**Comment by maxkfranz** [comment]
Rule of thumb : Rebase only on private branches.  Merge public branches.

E.g. rebase private `feature1` branch w.r.t. `development`:

```
git checkout feature1
git rebase development
```

E.g. merge `development` onto public `feature2` branch:


```
git checkout feature2
git merge development
```

**Comment by jvwong** [comment]
I thought I was rebasing my fork's branch. How to revert?

**Comment by maxkfranz** [comment]
It's simpler to just create a new branch off of dev and cherry pick the commits you want in the PR in the new branch.

**Comment by d2fong** [comment]
I think this might work?
git checkout < git commit id from before>
git checkout push -f <your branch name>

**Comment by jvwong** [comment]
I'm just cherry picking onto a new branch --> PR

**Pull request closed without merging**